### PR TITLE
Harris

### PR DESCRIPTION
**Pull request by simmplecoder**
### Description

Implement Harris corner detection, but without suggested result filtering. The algorithm writes Harris responses into `gray32_view_t` and the user can apply their chosen filtering/suppression algorithms. Loosening filtering will give edge detector.

### Environment

Everything compatible with Boost 1.70

### References

https://en.wikipedia.org/wiki/Harris_Corner_Detector

https://en.wikipedia.org/wiki/Structure_tensor

https://en.wikipedia.org/wiki/Image_derivatives

### Tasklist

- [X] Implement Harris corner detection algorithm
- [X] Extract contents of current test file into demo
- [ ] Add test cases
- [ ] Write docs that would be readable for most people
- [ ] Ensure all CI builds pass
- [ ] Review and approve

### CODE REVIEW

File: include/boost/gil/image_processing/harris.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#ifndef BOOST_GIL_HARRIS_HPP
+#define BOOST_GIL_HARRIS_HPP
+
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/typedefs.hpp>
+
+namespace boost { namespace gil {
+
+void find_corners_harris(
```
Comment (stefanseefeld) [inline include/boost/gil/image_processing/harris.hpp]: Can you add inline comments (doxygen-style, preferably) to explain (in broad strokes) what the function is doing, and specifically, what the arguments represent ?

Comment (simmplecoder) [reply include/boost/gil/image_processing/harris.hpp]: I was thinking about writing a bigger document, as this requires quite a lot of math to understand.

Comment (stefanseefeld) [reply include/boost/gil/image_processing/harris.hpp]: Both are needed: a design document that explains the algorithms, as well as inline documentation that explains the API. To be able to review the PR I need to have a basic understanding at least of what the arguments (and return values) are.

Comment (simmplecoder) [reply include/boost/gil/image_processing/harris.hpp]: Is it okay now? I will start moving the demo out of test file and adding simple tests for now. Will think about more complex ones later.

Comment (stefanseefeld) [reply include/boost/gil/image_processing/harris.hpp]: Thanks ! (You should build the docs (run `b2` inside the `doc/` subdirectory) to validate the formatting. I suspect the empty line between the comment block and the function definition prevents doxygen from parsing the comment as the function's documentation. But that's just from a casual look at the code.
The information itself is certainly useful. Thanks for adding it !

**Pull request closed without merging**